docs/book/src/topics/managedcluster.md

```diff
@@ -371,6 +392,32 @@ spec:
     enablePrivateClusterPublicFQDN: false # Allowed only when enablePrivateCluster is true
 ```
 
+## Immutable fields for Managed Clusters (AKS)
+
+Some fields from the family of Managed Clusters CRD are immutable. Which means 
+those can only be set during the creation time. 
+
+Following is the list of immutable fields for managed clusters:
+
+| CRD                      | jsonPath                           | Comment                   |
+|--------------------------|------------------------------------|---------------------------|
+| AzureManagedControlPlane | .spec.subscriptionID               |                           |
+| AzureManagedControlPlane | .spec.resourceGroupName            |                           |
+| AzureManagedControlPlane | .spec.nodeResourceGroupName        |                           |
+| AzureManagedControlPlane | .spec.location                     |                           |
+| AzureManagedControlPlane | .spec.sshPublicKey                 |                           |
+| AzureManagedControlPlane | .spec.dnsServiceIP                 |                           |
+| AzureManagedControlPlane | .spec.networkPlugin                |                           |
+| AzureManagedControlPlane | .spec.networkPolicy                |                           |
+| AzureManagedControlPlane | .spec.loadBalancerSKU              |                           |
+| AzureManagedControlPlane | .spec.apiServerAccessProfile       | except AuthorizedIPRanges |
+| AzureManagedMachinePool  | .spec.sku                          |                           |
+| AzureManagedMachinePool  | .spec.osDiskSizeGB                 |                           |
+| AzureManagedMachinePool  | .spec.osDiskType                   |                           |
+| AzureManagedMachinePool  | .spec.taints                       |                           |
```

Comment:
@zmalik @CecileRobertMichon on another workstream I ran into the fact that our AKS test template ships with taints, and I observed that I wasn't able to remove them (as the docs state they are immutable).

How confident are we that making taints immutable is the correct approach? It seems to me that you should be able manage AKS node pool taints via AzureManagedMachinePool.

cc @mtougeron @luthermonson @karthikbalasub @michalno1 @NovemberZulu @dkoshkin for thoughts here

Reply:
If they're not immutable on the AKS side then I don't think they should be immutable on the CAPZ side. However, it'll need a nice big callout in the docs to highlight that it won't change existing, running nodes in the node pool.